modules/internal/ChapelBase.chpl

```diff
@@ -1752,11 +1752,17 @@ module ChapelBase {
     if (arg != nil) {
       arg!.deinit();
 
-      on arg do
-        chpl_here_free(__primitive("_wide_get_addr", arg));
+      var addr = __primitive("_wide_get_addr", arg):c_void_ptr;
+      delete_helper(addr, arg.locale);
     }
   }
 
+  inline proc delete_helper(addr: c_void_ptr, loc: locale) {
```

Comment:
Nit, but I'd probably change the arg order to be locale, addr just since that's the order of wide pointers.

Reply:
And maybe I'm missing something but given that this is inlined it's not obvious to me how this reduces code size. Would we have a similar impact from just changing the `on arg do` to `on arg.locale do`?

Reply:
You're right that the inline helper doesn't really help. But the key is there shouldn't be anything generic in the `on` statement body. I think `on arg.locale` is optional:

```chapel
const addr = __primitive("_wide_get_addr", arg):c_void_ptr;
on arg.locale do // I think `on arg` is also fine here, but I'd use `on arg.locale` anyways.
  chpl_here_free(addr);
```